mart/attack/adversary.py

```diff
@@ -164,11 +164,13 @@ def configure_gradient_clipping(
                 self.gradient_modifier(group["params"])
 
     @silent()
-    def forward(self, *, batch: torch.Tensor | list | dict, model: Callable):
-        # TODO: We may see different batch format when attacking models outside MART. Add a batch_transform?
-        input, target = batch
+    def fit(self, *, batch: torch.Tensor | list | dict, model: Callable):
+        # Extract and canonicalize input/target so that is convenient for Adversary.
+        input, target = self.batch_c15n(batch)
 
-        # The attack needs access to the model at every iteration.
+        # Canonical form of batch in the adversary's optimization loop.
+        # We only see the canonicalized input/target in the attack optimization loop.
+        # The attack also needs access to the model at every iteration.
         batch_and_model = (input, target, model)
```

Comment:
I think you can change this to `batch_and_model = (batch, model)` instead now?

Reply:
Fixed in [52d3b23](https://github.com/IntelLabs/MART/pull/215/commits/52d3b23174b5e34fb5b18639fb845477e939bae3)